File: Labs/Lab4/python/figure.py
from abc_figure import ABC_Figure
from string import ascii_uppercase

class Figure(ABC_Figure):
    def __init__(self, *args):
        if not args:
            self.__cords = {}
        elif all(type(x) == tuple and len(x) == 2 and all(type(cord) == int for cord in x) for x in args):
            cords_amount = len(args)
            
            self.__cords = {ascii_uppercase[i]:args[i] for i in range(cords_amount)}
        elif len(args) == 1 and type(args[0]) == Figure:   
            self.__cords = args[0].get_cords()
        else:
            try:
                self.__cords = args[0].get_cords()
            except:
                print("Something wrong")
    
    def get_cords(self):
        return self.__cords
    
    def get_side(self, side1, side2):
        try:
            return ((side1[0] - side2[0])**2 + (side1[1] - side2[1])**2)**0.5
        except:
            raise Exception("Invalid points provided")
        
    
    
    

File: Labs/Lab4/c#/Figure.cs
using System.Collections.Generic;


namespace FiguresApp
{
    public class Figure
    {
        private Dictionary<char, (int, int)> _cords;

        public Figure()
        {
            _cords = new Dictionary<char, (int, int)>();
        }

        public Figure(params (int, int)[] points)
        {
            _cords = points.Select((p, i) => new { Index = i, Point = p })
                           .ToDictionary(item => (char)('A' + item.Index), item => item.Point);
        }
        public Figure(Figure other)
        {
            _cords = new Dictionary<char, (int, int)>(other.GetCords());
        }

        public Dictionary<char, (int, int)> GetCords()
        {
            return new Dictionary<char, (int, int)>(_cords);
        }

        public double GetSide(char point1, char point2)
        {
            if (_cords.ContainsKey(point1) && _cords.ContainsKey(point2))
            {
                var side1 = _cords[point1];
                var side2 = _cords[point2];
                return Math.Sqrt(Math.Pow(side1.Item1 - side2.Item1, 2) + Math.Pow(side1.Item2 - side2.Item2, 2));
            }
            throw new ArgumentException("Invalid points provided");
        }
    }
}